src/seeds.harvest.cpp

```diff
@@ -905,3 +910,278 @@ ACTION harvest::setorgtxpt(name organization, uint64_t tx_points) {
+
+
+void harvest::calcmintrate () {
+  require_auth(get_self());
+
+  uint64_t day = utils::get_beginning_of_day_in_seconds();
+  auto previous_day_temp = eosio::time_point_sec((day - (3 * utils::moon_cycle)) / 86400 * 86400);
+  uint64_t previous_day = previous_day_temp.utc_seconds;
+
+  auto current_qev_itr = monthlyqevs.get(day, "no monthly qev found");
+  auto previous_qev_itr = monthlyqevs.get(previous_day, "no previous monthly qev found");
+
+  double volume_growth = double(current_qev_itr.qualifying_volume - previous_qev_itr.qualifying_volume) / previous_qev_itr.qualifying_volume;
+
+  int64_t target_supply = (1.0 + volume_growth) * previous_qev_itr.circulating_supply;
+
+  int64_t delta = target_supply - current_qev_itr.circulating_supply;
+
+  double mint_rate = delta / 708.0;
+
+  auto mitr = mintrate.begin();
+  if (mitr != mintrate.end()) {
+    mintrate.modify(mitr, _self, [&](auto & item){
+      item.mint_rate = mint_rate;
+      item.volume_growth = volume_growth * 10000;
+      item.timestamp = eosio::current_time_point().sec_since_epoch();
+    });
+  } else {
+    mintrate.emplace(_self, [&](auto & item){
+      item.id = mintrate.available_primary_key();
+      item.mint_rate = mint_rate;
+      item.volume_growth = volume_growth * 10000;
+      item.timestamp = eosio::current_time_point().sec_since_epoch();
+    });
+  }
+
+}
+
+
+uint64_t harvest::config_get(name key) {
+  auto citr = config.find(key.value);
+  if (citr == config.end()) { 
+    check(false, ("settings: the "+key.to_string()+" parameter has not been initialized").c_str());
+  }
+  return citr->value;
+}
+
+void harvest::send_distribute_harvest (name key, asset amount) {
+
+  cancel_deferred(key.value);
+
+  action next_execution(
+    permission_level{get_self(), "active"_n},
+    get_self(),
+    key,
+    std::make_tuple(uint64_t(0), config_get("batchsize"_n), amount)
+  );
+
+  transaction tx;
+  tx.actions.emplace_back(next_execution);
+  tx.delay_sec = 1;
+  tx.send(key.value, _self);
+
+}
+
+void harvest::withdraw_aux (name sender, name beneficiary, asset quantity, string memo) {
+  token::transfer_action t_action{contracts::token, { sender, "active"_n }};
+  t_action.send(sender, beneficiary, quantity, memo);
+}
+
+void harvest::runharvest() {
+  require_auth(get_self());
+
+  auto mitr = mintrate.begin();
+  check(mitr != mintrate.end(), "mint rate table is empty");
+
+  if (mitr -> mint_rate <= 0) { return; }
+
+  asset quantity = asset(mitr -> mint_rate, test_symbol);
+  string memo = "harvest";
+
+  print("mint rate:", quantity, "\n");
+
+  token::issue_action_test t_issue{contracts::token, { contracts::token, "minttst"_n }};
+  t_issue.send(get_self(), quantity, memo);
+
+  double users_percentage = config_get("hrvst.users"_n) / 1000000.0;
+  double bios_percentage = config_get("hrvst.bios"_n) / 1000000.0;
+  double orgs_percentage = config_get("hrvst.orgs"_n) / 1000000.0;
+  double global_percentage = config_get("hrvst.global"_n) / 1000000.0;
+
```

Comment:
I think we need to add a "escrow" percentage also - some of the Seeds are supposed to get paid out with escrow but I am not sure yet how that really works. Since escrow is based on people... not everyone gets the same...  IDK

No action item, just thinking out loud here. We can release this as is here.